decompiler/frontend/binaryninja/handlers/constants.py

```diff
@@ -29,72 +14,71 @@ def register(self):
+                    - lift as datavariable
+                3. Address has a symbol, which is not a datasymbol
+                    - lift as symbol
+                4. Address has a function there 
+                    - lift the function symbol as symbol
+                5. Address has a datasymbol or void/void* datavariable or None there 
+                    - lift as char* if there is a string, otherwise as void* with raw bytes (if the datavariable was a ptr, lift as &ptr*, otherwise as ptr*)
 
-    @staticmethod
-    def _get_symbol(bv: BinaryView, address: int) -> Optional[bSymbol]:
-        """Retrieve the symbol at the given location, if any."""
-        if symbol := bv.get_symbol_at(address):
-            return symbol
-        elif function := bv.get_function_at(address):
-            return function.symbol
-        return None
+                    - there were symbols which call them self recursively, therefore a small check before the end
+        """
+        view = pointer.function.view
+
+        if pointer.constant == 0:
+            return Constant(pointer.constant, vartype=Integer.uint64_t() if view.address_size == 8 else Integer.uint32_t())
+
+        if (variable := view.get_data_var_at(pointer.constant)) and not (isinstance(variable.type, PointerType) and isinstance(variable.type.target, VoidType)):
+            return self._lifter.lift(variable, view=view, parent=pointer)
+
+        if (symbol := view.get_symbol_at(pointer.constant)) and symbol.type != SymbolType.DataSymbol:
+            return self._lifter.lift(symbol)
+
+        if function := view.get_function_at(pointer.constant):
+            return self._lifter.lift(function.symbol)
+
+        var_ref_string = (view.get_string_at(variable.value, True) or view.get_ascii_string_at(variable.value, min_length=2)) if variable and variable.value else None
+        var_ref_value = view.get_data_var_at(variable.value) if variable and variable.value else None
+
+        if var_ref_value and pointer.constant == var_ref_value.address: # Recursive ptr to itself (0x4040 := 0x4040), lift symbol if there, else just make a data_addr symbol
+            data_symbol =  view.get_symbol_at(variable.value)
+            var_ref_value = data_symbol if data_symbol else Symbol("data_" + f"{pointer.constant:x}", pointer.constant, vartype=Integer.uint32_t())    
+
+        g_var = GlobalVariable(
+            name=symbol.name[:-2] + "_" + view.get_sections_at(variable.address)[0].name[1:] if symbol and symbol.name.find(".0") != -1 \
+                else symbol.name if symbol else "data_" + f"{pointer.constant:x}",
+            vartype=self._lifter.lift(Type.pointer(view.arch, Type.char())) if var_ref_string else self._lifter.lift(Type.pointer(view.arch, Type.void())),
+            ssa_label=pointer.ssa_memory_version if pointer else 0,
+            initial_value=self._lifter.lift(var_ref_value, view=view, parent=pointer) if var_ref_value else Constant(var_ref_string.value) \
```

Comment:
Where exactly? 

Every usage of `variable.value` checks if `variable` and `variable.value` are valid.
In the if clause with `var_ref_value` we don't need to check for `variable`, because  `var_ref_value` does not exists without it.

Reply:
I'm sorry, I think I got lost on the two-line double ternary operation here - did not see the check in line 73. Never encountered something like this before

Reply:
Nah I'm sorry for this kind of code. 
Sadly this is still better, "then writing if/else" for every case, which would explode the function .

Reply:
Also done